torchvision/csrc/ops/quantized/cpu/qroi_align_kernel.cpp

```diff
@@ -0,0 +1,201 @@
+#include <ATen/ATen.h>
+#include <ATen/native/quantized/affine_quantizer.h>
+#include <torch/library.h>
+
+#include "../../roi_align.h"
+
+namespace vision {
+namespace ops {
+
+namespace {
+
+template <typename T>
+void qroi_align_forward_kernel_impl(
+    int n_rois,
+    const at::Tensor& t_input,
+    const float& spatial_scale,
+    int channels,
+    int height,
+    int width,
+    int pooled_height,
+    int pooled_width,
+    int sampling_ratio,
+    bool aligned,
+    const at::Tensor& t_rois,
+    T* output) {
+  const T* input = t_input.contiguous().data_ptr<T>();
+  int64_t input_zp = t_input.q_zero_point();
+  float input_scale = t_input.q_scale();
+
+  const T* rois = t_rois.contiguous().data_ptr<T>();
+  int64_t rois_zp = t_rois.q_zero_point();
+  float rois_scale = t_rois.q_scale();
+
+  for (int n = 0; n < n_rois; n++) {
+    int index_n = n * channels * pooled_width * pooled_height;
+
+    const T* offset_rois = rois + n * 5;
+    int roi_batch_ind = at::native::dequantize_val(
+        rois_scale, rois_zp, offset_rois[0]); // FIXME: This can be out of the
+                                              // range of the quantized type!!
```

Comment:
I'm not sure what to do with that: the first column of the rois is supposed to be the batch index, but it might be impossible to properly represent the batch index, e.g.:

- if there are 256 batches on a qint8 tensor
- we can't represent 1 with (scale, zero_point) = (2, 0)